### DISCUSSION TITLE
MAUI support

### DISCUSSION THREAD

**Discussion by BCaruso515** [QuestPDF/QuestPDF#1078] [General]
Has anyone been able to implement QuestPDF in .NET Maui IOS or Android? It works fine on Windows but when deploying to IOS or Android I'm getting the following error... "the type initializer for questpdf.settings threw an exception" on the QuestPDF.Settings.License = LicenseType.Community; line.

**Comment by girlpunk** [comment]
Please could you post the full exception. However, it's likely that QuestPDF hasn't been compiled for the type of CPU architecture you're trying to run on.

**Reply by BCaruso515** [reply]
I assumed it may be a compatibility issue. I was just hoping someone may have been able to make it work.

[0:] An error occurred: 'QuestPdfSkia'. Callstack: '   at QuestPDF.Skia.SkNativeDependencyCompatibilityChecker.ExecuteNativeCode()
   at QuestPDF.Helpers.NativeDependencyCompatibilityChecker.CheckIfExceptionIsThrownWhenLoadingNativeDependencies()'

**Comment by SoyPan** [comment]
Are you using a 2023.12.X version? Support for those platforms was dropped after that version.

**Reply by BCaruso515** [reply]
Version=2024.10.0.0
I've been able to create my reports with iText.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

As @SoyPan correctly noted, support for mobile platforms was dropped in version 2024.3.0 when QuestPDF transitioned to its own native Skia layer. QuestPDF does not currently support Xamarin, MAUI, Blazor WASM, or Uno, and there are no immediate plans to reintroduce support for these platforms. As PDF generation is primarily a backend-specific feature, we recommend generating documents on the server side. We may consider reintroducing support for mobile platforms depending on community demand, but it remains low in priority for the foreseeable future.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
